hasher-matcher-actioner/hmalib/models.py

```diff
@@ -281,3 +309,75 @@ def from_time_range(
             & Key("UpdatedAt").between(start_time, end_time),
             ProjectionExpression=cls.DEFAULT_PROJ_EXP,
         ).get("Items", [])
+
+
+@dataclass
+class MatchMessage(SNSMessage):
+    """
+    Captures a set of matches that will need to be processed. We create one
+    match message for a single content key. It is possible that a single content
+    hash matches multiple datasets. When it does, the entire set of matches are
+    forwarded together so that *one* appropriate action can be taken.
+
+    - `content_key`: A way for partners to refer uniquely to content on their
+      site
+    - `content_hash`: The hash generated for the content_key
+    """
+
+    content_key: str
+    content_hash: str
+    match_details: t.List["DatasetMatchDetails"] = field(default_factory=list)
+
+    def to_sns_message(self) -> str:
+        return json.dumps(
+            {
+                "ContentKey": self.content_key,
+                "ContentHash": self.content_hash,
+                "MatchDetails": [x.to_dict() for x in self.match_details],
+            }
+        )
+
+    @classmethod
+    def from_sns_message(cls, message: str) -> "MatchMessage":
+        parsed = json.loads(message)
+        return cls(
+            parsed["ContentKey"],
+            parsed["ContentHash"],
+            [DatasetMatchDetails.from_dict(d) for d in parsed["MatchDetails"]],
+        )
+
+
+@dataclass
+class DatasetMatchDetails:
+    """
+    Dataset fields:
+    - `banked_content_id`: Inside the bank, what's a unique way to refer to what
+      was matched against?
+    - `bank_id`: [optional][Defaults to 'threatexchange_all_collabs'] Which bank
+      did we fetch this banked_content from?
+    - `bank_source`: [optional][Defaults to 'api/threatexchange'] This is
+      forward looking, but potentially, we could have this be 'local', or
+      'api/some-other-api'
+    """
+
+    banked_indicator_id: str
```

Comment:
Don't feel strongly about it but indicator is pretty source specific. I don't love the sound of`banked_signal_id` but if we wanted to match the signal_source pattern in PDQMatchRecord that's what I'd go with.

Reply:
We can do what mms does and just call everything under the sun cluster! (please don't do this).

We will eventually go content-centric even on banking, and so submit `banked_content_id`, which will parallel `content_id` on the pipeline.